docs/resie_energy_systems.md

```diff
@@ -102,8 +94,37 @@ Currently, ReSiE is based on the following units:
+
+### Transformer chains
+
+With multiple transformers in a chain there arises a problem in calculating the energy each transformer can process, as this depends on the energy the others can process. This is the main purpose behind the `potential` operation, which only exists for transformers and is used in a simple algorithm to solve the problem:
+
+1. The last transformer in a chain, where last refers to not outputting to another transformer, calculates its energy potentials assuming that any input or output that has not been determined yet is infinite.
+2. This repeats for the other transformers going to the front of the chain, where front refers to not having a transformer as an input.
+3. The transformer at the front now performs the `process` step, which is possible as now the whole energy system is specified in regards to energy potentials.
+4. This repeats for the other transformers going to the back of the chain, however each transformer recalculates its own potentials because the inputs will have changed from the previous assumed-as-infinite values.
+
+This back-and-forth approach works as it communicates the non-infinite demands and supplies from non-transformers across the chain and then recalculates with the correct `process` values once all potentials are specified.
+
+**Note:** At the moment this does not work if transformers are connected across one or more busses, because this is not recognised as a chain. This will be addressed in a future version of ReSiE.
+
+### Bus chains
+
+Because busses are both used as an abstraction over the actual hydraulical and electrical network of an energy system, as well as the component for distributing energy in a sub-network of the energy system, it often makes sense from a modelling perspective to have multiple busses of the same medium connected to each other, forming a chain. For example an energy system of a district might provide heat in a central bus that outputs to other busses in each building, which have local heat storages and other components.
+
+Given that energy can only flow in one direction in such a chain, the entire chain can be replaced by a single bus, which has all non-bus inputs and outputs as its own inputs and outputs. This is called a proxy bus, with the original busses being called the principals. During the construction of the proxy bus, the energy flow matrices of the principals, as well as the direction of energy flow between busses is taken into consideration to determine which inputs of the proxy bus can provide energy for which outputs and in which order this is happening.
```

Comment:
Add explanation:
These proxy busses are created from single, directly interconnected busses within the preprocessing of ReSiE in order to allow easier calculation within the computational core. For the outputs, these proxy busses are then converted back to the original pricipal busses. During the construction...